msteams-platform/toolkit/toolkit-v4/debug-your-Teams-app-test-tool-vs.md

```diff
@@ -0,0 +1,227 @@
+    
+    1. Replace model name with model deployment name in `Program.cs` file.
+    
+         :::image type="content" source="../../assets/images/teams-toolkit-v2/teams-toolkit-vs/azure-openai-model-deployment-name.png" alt-text="Screenshot shows the AzureOpenAI model deployement name updated.":::
+    ---
+
+1. In the debug dropdown list, select **Teams App Test Tool (browser)**.
+
+   :::image type="content" source="../../assets/images/teams-toolkit-v2/teams-toolkit-vs/app-test-tool.png" alt-text="Screenshot shows the option to select the Teams app test tool from the dropdown list.":::
+
+1. Test Tool opens the bot in a webpage.
+
+   :::image type="content" source="../../assets/images/teams-toolkit-v2/teams-toolkit-vs/test-tool.png" alt-text="Screenshot shows the bot opens the test tool in web page." lightbox="../../assets/images/teams-toolkit-v2/teams-toolkit-vs/debug-test-tool-vs.png"::: 
+
+## Activity triggers
+
+You can mock an activity in Test Tool using activity triggers. There are two types of activity triggers:
+
+* [Predefined activity triggers](#predefined-activity-triggers)
+* [Custom activity triggers](#custom-activity-triggers)
+
+### Predefined activity triggers
+
+Test Tool provides predefined activity triggers to test the functionalities of your bot.
+
+| Category | Activity | Handler |
+| --- | --- | --- |
+| Trigger Installation Update Activity | Install bot <br><br><br> Uninstall bot | `onInstallationUpdate` <br> `onInstallationUpdateAdded` <br><br> `onInstallationUpdate` <br> `onInstallationUpdateRemove`|
+| Trigger Conversation Update Activity | Add user <br><br> Add bot <br><br> Add channel | `onMembersAdded`<br><br> `onTeamsMembersAddedEvent` <br><br> `onTeamsChannelRenamedEvent` |
+| | Remove user <br><br><br> Remove bot <br><br><br> Remove channel <br><br> Remove team | `onMembersRemoved` <br> `onTeamsMembersRemovedEvent` <br><br> `onMembersRemoved` <br> `onTeamsMembersRemovedEvent` <br><br> `onTeamsChannelDeletedEvent` <br><br> `onTeamsTeamDeletedEvent` |
+| | Rename channel <br><br> Rename team | `onTeamsChannelRenamedEvent` <br><br> `onTeamsTeamRenamedEvent` |
+
+> [!NOTE]
+> All types of activities aren't available in all scopes. For example, you can't add or remove a channel in a personal chat or a group chat.
+
+Predefined activity triggers are available in the **Mock an Activity** menu in Test Tool.
+
+To mock an **Add user** activity, follow these steps:
+
+1. In Visual Studio Code, go to **Solution Explorer**.
+1. Select **Program.cs** file.
+1. In the **Program.cs** file under `builder.Services.AddTransient<IBot>(sp =>`, add the following code:
+
+    
+    ```json
+    app.OnConversationUpdate("membersAdded", async (context, state, cancellationToken) =>
+    {
+    await context.SendActivityAsync($"new member added", cancellationToken: cancellationToken);
```

Comment:
please add indent here and below line for better code style.

Reply:
Updated.